addons/mass_mailing/models/mass_mailing.py

```diff
@@ -316,6 +316,18 @@ def _get_source_id(self, cr, uid, context=None):
     def _get_medium_id(self, cr, uid, context=None):
         return self.pool['ir.model.data'].xmlid_to_res_id(cr, uid, 'utm.utm_medium_email')
 
+    def create(self, cr, uid, vals, context=None):
+        if not vals['campaign_id']:
+            utm_campaign_id = self.pool['utm.campaign'].create(cr, uid, {'name': vals['name']}, context=context)
+            vals['campaign_id'] = utm_campaign_id
+        return super(MassMailingCampaign, self).create(cr, uid, vals, context=context)
+
+    def write(self, cr, uid, ids, vals, context=None):
+        mass_mailing_campaign = self.browse(cr, uid, ids, context=context) 
```

Comment:
Actually the best thing to do should probably to avoid having a name field defined on mass_mailing.campaign model. Indeed the name field could come from utm.campaign. Probably a development to do in master. So here this seems correct for a stable version.

the "if campaign.campaign_id" is not necessary as campaign_id is required. And to be exact, you should check that campaign_id is not in vals (do not update something that won't refer the mailing campaign).

By the way, your code is not batch-friendly.